### ISSUE TITLE
AWS: No error when using an incorrect "ssl_certificate_id" in aws_elb

### ISSUE THREAD

**Issue by jmprusi** [hashicorp/terraform#1627]
There's no error message when using an incorrect "ssl_certificate_id" so terraform, when doing a "plan" just keeps showing that the elb needs to be modified.

When doing a "terraform apply", there is no error displayed, everything seems to have worked.

**Comment by mitchellh** [comment]
The plan wouldn't know, but the apply should fail if the AWS API returns an error. We'll take a look. Thanks.

**Comment by catsby** [comment]
Hey @jmprusi – are you using `https` for the `lb_protocol`? From my experiemnts, AWS is simply ignoring the `ssl_certificate_id` field if you're _not_ using `https` as the protocol, without any warning/error. 

If you _are_ using `https`, it throws an error for me if I use an invalid / incorrect ssl cert id:

```
* Failure adding new or updated ELB listeners: CertificateNotFound: Server Certificate not found for the key: arn:was:iam:<bla>
	status code: 400, request id: [cab8de7d-ff2a-11e4-9e1f-f50fab7b39f2]
```

**Comment by jmprusi** [comment]
Hi @catsby, I can't get to reproduce the error again (with version 5.0.1), although I don't have the code that triggered the issue, I've tried:

* Creating a new elb with an incorrect ssl_certificate_id  = error
* Creating a new elb without a SSL listener, and then adding a SSL listener to it with an incorrect ssl_certificate_id = error

I think that I was using the https protocol because when I noticed the typo in the ssl_certificate_id, fixing it solved the problem (instead of "arn:aws:iam.." was "rn:aws:iam.." )

Sorry..

**Comment by catsby** [comment]
Thanks for getting back to me @jmprusi – with that information, I'm going to assume this is fixed with the latest release (0.5.2 at the time). 

Thanks for opening the issue! Let me know if something pops up again
